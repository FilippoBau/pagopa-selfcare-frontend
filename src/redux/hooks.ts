import {TypedUseSelectorHook, useDispatch, useSelector} from 'react-redux';
import {useHistory} from 'react-router-dom';
import type {RootState, AppDispatch} from './store';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

<<<<<<< HEAD
export const useAppSelectorWithRedirect = ({
  selector,
  routeToRedirect,
  conditionToRedirect,
}: {
  selector: (state: RootState) => any;
  routeToRedirect?: string;
  conditionToRedirect?: (value: any) => boolean;
}) => {
  const selectedReduxState = useAppSelector(selector);
  const history = useHistory();
  const conditionResults = conditionToRedirect ? conditionToRedirect(selectedReduxState) : false;
  if (
    routeToRedirect &&
    (!selectedReduxState ||
      (typeof selectedReduxState === 'object' && Object.keys(selectedReduxState).length === 0) ||
      conditionResults)
  ) {
    history.push(routeToRedirect);
  }
  return selectedReduxState;
=======
export const useAppSelectorWithRedirect = (
    selector: (state: RootState) => any,
    routeToRedirect?: string
) => {
    const selectedReduxState = useAppSelector(selector);
    const history = useHistory();
    if (
        routeToRedirect &&
        (!selectedReduxState ||
            (typeof selectedReduxState === 'object' && Object.keys(selectedReduxState).length === 0))
    ) {
        history.push(routeToRedirect);
    }
    return selectedReduxState;
>>>>>>> 3f32cfc3 (Formatting (#542))
};
