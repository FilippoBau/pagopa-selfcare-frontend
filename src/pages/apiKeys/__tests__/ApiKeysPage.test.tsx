import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../../../redux/store';
import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { theme } from '@pagopa/mui-italia';
import { ThemeProvider } from '@mui/system';
import ApiKeysPage from '../ApiKeysPage';
import routes from '../../../routes';

//SNAPSHOT TESTING
it('renders correctly', () => {
  const tree = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={[routes.APIKEYS]}>
        <Route path={routes.APIKEYS}>
          <ThemeProvider theme={theme}>
            <ApiKeysPage />
          </ThemeProvider>
        </Route>
      </MemoryRouter>
    </Provider>
  );
  expect(tree).toMatchSnapshot();
});
