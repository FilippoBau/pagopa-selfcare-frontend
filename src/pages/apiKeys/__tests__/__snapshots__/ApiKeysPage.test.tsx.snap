// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-4uxo5g-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1539t99-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-jeqi4w"
            >
              <div
                class="MuiBox-root css-ir1t2y"
              >
                <ul
                  class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                >
                  <div
                    class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1k5qf7y-MuiButtonBase-root-MuiListItemButton-root"
                    data-testid="home-test"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root css-1vo01mv-MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-ycncy1-MuiSvgIcon-root-MuiIcon-root"
                        data-testid="DashboardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root css-mmjwmt-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1gq3qqz-MuiTypography-root"
                      >
                        sideMenu.home.title
                      </span>
                    </div>
                  </div>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-13qir85-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-v4b62z"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-14b6npa-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13e0yuj-MuiGrid-root"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4 css-tp226b-MuiTypography-root"
                >
                  apiKeysPage.title
                </h4>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-kxmh4q-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1mx152z-MuiTypography-root"
                >
                  apiKeysPage.subtitle
                </p>
              </div>
            </div>
            <div
              class="MuiBox-root css-hsy21v"
            >
              <div
                class="MuiBox-root css-0"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-3grx0y-MuiTypography-root"
                >
                  apiKeysPage.decription
                </h6>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-9l2l0u-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
                      data-testid="AddIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                  apiKeysPage.apiNotPresent.buttonLabel
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiBox-root css-rg3i8g"
            >
              Non Ã¨ stata ancora generata nessuna chiave API per questo ente.
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1lr2oxb-MuiTypography-root-MuiLink-root"
              >
                <strong>
                   Genera API Key
                </strong>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-4uxo5g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1539t99-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-jeqi4w"
          >
            <div
              class="MuiBox-root css-ir1t2y"
            >
              <ul
                class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              >
                <div
                  class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1k5qf7y-MuiButtonBase-root-MuiListItemButton-root"
                  data-testid="home-test"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root css-1vo01mv-MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-ycncy1-MuiSvgIcon-root-MuiIcon-root"
                      data-testid="DashboardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root css-mmjwmt-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1gq3qqz-MuiTypography-root"
                    >
                      sideMenu.home.title
                    </span>
                  </div>
                </div>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-13qir85-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-v4b62z"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-14b6npa-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13e0yuj-MuiGrid-root"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 css-tp226b-MuiTypography-root"
              >
                apiKeysPage.title
              </h4>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-kxmh4q-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1mx152z-MuiTypography-root"
              >
                apiKeysPage.subtitle
              </p>
            </div>
          </div>
          <div
            class="MuiBox-root css-hsy21v"
          >
            <div
              class="MuiBox-root css-0"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-3grx0y-MuiTypography-root"
              >
                apiKeysPage.decription
              </h6>
            </div>
            <div
              class="MuiBox-root css-0"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-9l2l0u-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jxdcj3-MuiSvgIcon-root"
                    data-testid="AddIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                    />
                  </svg>
                </span>
                apiKeysPage.apiNotPresent.buttonLabel
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root css-rg3i8g"
          >
            Non Ã¨ stata ancora generata nessuna chiave API per questo ente.
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1lr2oxb-MuiTypography-root-MuiLink-root"
            >
              <strong>
                 Genera API Key
              </strong>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
